File: Resueltos por competidor/Samuel Mesa/cpp/TrainSwaping.cpp
#include <bits/stdc++.h>
using namespace std;

int main(int argc, char *argv[]) {
	int casos,nums,cont,cambio;
	cin>>casos;
	for(int i=0;i<casos;i++){
		cin>>nums;
		int arr[nums];
		cont=0;
		for(int j=0;j<nums;j++){
			cin>>arr[j];
		}
		for(int k=0;k<nums;k++){
			for(int h =0;h<nums-1;h++){
                    if(arr[h]>arr[h+1]){
					cambio =arr[h];
					arr[h]=arr[h+1];
					arr[h+1]=cambio;
					cont++;
				}
			}
		}
		cout<<"Optimal train swapping takes "<<cont<<" swaps."<<endl;
	}
	
	return 0;
}

File: Resueltos por competidor/William Castillo/TrainSwaping.java
import java.util.Iterator;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int as = sc.nextInt();
		for (int i = 0; i < as; i++) {
			int cont = 0;
			int casos = sc.nextInt();
			int arr[] = new int[casos];
			for (int j = 0; j < casos; j++) {
				int nums = sc.nextInt();
				arr[j] = nums;
			}
			for (int k = 0; k < casos; k++) {
				for (int j = 0; j < casos - 1; j++) {
					if (arr[j] > arr[j + 1]) {
						int cambio = arr[j];
						arr[j] = arr[j + 1];
						arr[j + 1] = cambio;
						cont++;
					}
				}

			}

			System.out.println("Optimal train swapping takes " + cont + " swaps.");
			cont=0;
			
		}

	}

}
